src/models/environment.ts

```diff
@@ -45,35 +44,62 @@ export abstract class Environment extends Data implements IResourceBase {
   kafkaClusters!: KafkaCluster[];
   schemaRegistry?: SchemaRegistry | undefined;
 
+  // updated by the ResourceViewProvider from connectionLoading/connectionUsable events
+  isLoading: boolean = false;
+
   get hasClusters(): boolean {
     return this.kafkaClusters.length > 0 || !!this.schemaRegistry;
   }
 }
 
+export interface CCloudEnvironmentProps {
+  id: string;
+  name: string;
+  streamGovernancePackage: string;
+  kafkaClusters: CCloudKafkaCluster[];
+  schemaRegistry?: CCloudSchemaRegistry | undefined;
+}
+
 /** A Confluent Cloud {@link Environment} with additional properties. */
 export class CCloudEnvironment extends Environment {
   readonly connectionId: ConnectionId = CCLOUD_CONNECTION_ID;
   readonly connectionType: ConnectionType = ConnectionType.Ccloud;
 
   readonly iconName: IconNames = IconNames.CCLOUD_ENVIRONMENT;
 
-  streamGovernancePackage!: Enforced<string>;
-  // set explicit CCloud* typing
-  kafkaClusters: CCloudKafkaCluster[] = [];
-  schemaRegistry: CCloudSchemaRegistry | undefined = undefined;
+  kafkaClusters: CCloudKafkaCluster[]; // explicitly typed
+  schemaRegistry?: CCloudSchemaRegistry | undefined; // explicitly typed
+  streamGovernancePackage: string;
+
+  constructor(props: CCloudEnvironmentProps) {
```

Comment:
Excessive interfaces for each class seems too verbose to my taste. If you want to express "this constructor should receive an object of own fields", you can do following:

```
class CCloudEnvironment extends Environment {
  constructor(props: Pick<CCloudEnvironment, "id" | "name" | "streamGovernancePackage" | "kafkaClusters" | "schemaRegistry">) {}
}

Reply:
Agreed those dedicated interfaces seemed pretty redundant; thanks for the suggestion to use `Pick`, really good to know!